circuit_maintenance_parser/provider.py

```diff
@@ -122,6 +123,9 @@ def get_maintenances(self, data: NotificationData) -> Iterable[Maintenance]:
             logger.debug("Skipping notification %s due filtering policy for %s.", data, self.__class__.__name__)
             return []
 
+        if os.getenv("OPENAI_TOKEN"):
+            self._processors.append(CombinedProcessor(data_parsers=[EmailDateParser, OpenAIParser]))
```

Comment:
because the "Stamp" (when the notification was received) it's not available in the notification body. This parser allows extracting the date from the notification without the need of ML processing.